src/app/components/Viewer/EBookViewer.jsx

```diff
@@ -40,15 +40,16 @@ class EBookViewer extends React.Component {
 
   render() {
     const { router } = this.context;
-    const work = this.props.location.state ? this.props.location.state.work : undefined;
+    const edition = this.props.location.state ? this.props.location.state.edition : undefined;
     const bookUrl = this.state.bookUrl;
 
     return (
       <span>
         <Breadcrumbs
           router={router}
           location={this.props.location}
-          workDetail={work}
+          workDetail={{ uuid: edition.work_uuid, title: edition.title }}
```

Comment:
This is just a note, edition titles and work titles can be different. We try to choose the most popular or original edition title for the work, but they can be different. It's usually not noticeable but it might be something that we want to correct in the future.

Reply:
Yeah it's ... tricky.  I did it this way to try not to pass in work into state (we don't always have work in state, (clicking "read online" from the Edition Detail page) but we also don't always have edition.title in state (clicking "read online" from the Search Results page).  It makes it possible that, depending on where you clicked "Read Online" from, you might get a different work name.   (Just like if you navigate to a reader page from a link someone sends you, you won't get work and edition in breadcrumbs)  This ... can't be helped, unless we have the "Read Online" page be an "Item" page where "Item" knows what work and edition belongs to.

Reply:
yeah, I think it's fine for it to work this way. It's actually even more complicated because the item you're viewing might actually have a different title from the work, so then the question becomes: which is the right title to display? So in short: definitely should just be left as is, I'm just trying to flag stuff like this so we know its out there.

Reply:
sounds good.  Books, huh.

Reply:
OK so right now the breadcrumbs will always use the work title that's in edition if there is one, but it will substitute with the work title if there isn't one.  That's happening in `EditionCard.getReadOnlineLInk`